pandas/tests/test_sorting.py

```diff
@@ -183,6 +185,22 @@ def test_nargsort(self):
         exp = list(range(5)) + list(range(105, 110)) + list(range(104, 4, -1))
         tm.assert_numpy_array_equal(result, np.array(exp), check_dtype=False)
 
+    @pytest.mark.parametrize('data', [
+        Categorical(['a', 'c', 'a', 'b']),
+        to_datetime([0, 2, 0, 1]).tz_localize('Europe/Brussels')])
+    def test_nargsort_extension_array(self, data):
+        result = nargsort(data)
+        expected = np.array([0, 2, 3, 1], dtype=np.intp)
+        tm.assert_numpy_array_equal(result, expected)
+
+    def test_nargsort_datetimearray_warning(self, recwarn):
+        # https://github.com/pandas-dev/pandas/issues/25439
+        # can be removed once the FutureWarning for np.array(DTA) is removed
```

Comment:
is there a reason you are not using ``tm.assert_produces_warning``, this looks non-standard

Reply:
Because I want to assert that a specific warning is *not* raised. I found it a bit clunky as well, so if you know a better way, all ears.

Reply:
`with pytest.raises(None):` is the cleanest, but I think our linter disallows that.

Reply:
Assuming you mean `pytest.warns(None)`: in that case you still have to do something like

```
with pytest.warns(None) as record:
    ....

assert len(record) == 0
```

giving basically the same as I did above (but above I did it with the recwarn fixture instead of the with statement. Happy to use the with statement if that is more typically used / easier to understand. Actually now thinking about it, I would find that clearer myself :-))

Reply:
Yes, I meant `pytest.warns`. I really thought that `pytest.warns(None)` asserted that there were no warnings, but apparently not.

In theory, you could rely on our global "fail on warning" setting from the CI, but I don't think I trust that enough :)

Reply:
``tm.assert_produces_warning(None)`` already does this

Reply:
But not for a *specific* warning, what I tried to do here.
But OK, maybe it is not that important to do it (the reason I wanted to do that was to avoid unrelated failures, eg when numpy starts raising a warning on something unrelated used under the hood).

Will change to `tm.assert_produces_warning(None)`

Reply:
> But not for a specific warning, what I tried to do here.
But OK, maybe it is not that important to do it (the reason I wanted to do that was to avoid unrelated failures, eg when numpy starts raising a warning on something unrelated used under the hood).

sure, but try to guard against *unexpected* warnings is a bit of a rabbit hole